charms/kfp-profile-controller/tests/integration/test_charm.py

```diff
@@ -35,19 +44,181 @@ async def test_build_and_deploy(ops_test: OpsTest):
         "minio:object-storage",
     )
 
-    # TODO: Need a better test for checking everything is ok
+    # Deploy charms responsible for CRDs creation
+    await ops_test.model.deploy(
+        entity_url="kubeflow-profiles",
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        entity_url="metacontroller-operator",
+        trust=True,
+    )
+
     # Maybe: await ops_test.model.wait_for_idle(raise_on_error=False, raise_on_blocked=True) ?
-    await ops_test.model.wait_for_idle(timeout=60 * 10)
-
-
-# TODO: Real testing requires:
-#  * kubeflow-profile controller, or something that will make namespaces/service accounts in a
-#    similar fashion (some resources deployed to each namespace by kfp-profile-controller use the
-#    `default-editor` service account, which is created by the main kubeflow-profile controller)
-#  * This charm deploying successfully is not a full test of function.  We need to create a new
-#    tracked namespace (namespace that has label pipelines.kubeflow.org/enabled=true) and ensure
-#    all expected resources are deployed and come up successfully (eg: if service account is not
-#    found, deployments will exist but pods will never be "ready")
-#  * check whether secrets and other things get updated based on the config sent to sync.py/kfppc
-#  * could create namespaces with the right pipelines label to trigger the sync, but also need to
-#    create a dummy poddefaults CRD too
+    await ops_test.model.wait_for_idle(status="active", timeout=60 * 10)
+
+
+async def test_status(ops_test: OpsTest):
+    charms = ("kubeflow-profiles", "metacontroller-operator", "kfp-profile-controller")
+    for charm in charms:
+        assert ops_test.model.applications[charm].units[0].workload_status == "active"
+
+
+async def test_profile_creation(lightkube_client, profile):
+    # Test whether a namespace was created for this profile
```

Comment:
This makes me think we're testing whether the profile was created, but I think the intent is to test whether a `Namespace`, etc, was created for the given `Profile`?  Maybe reword (and make the headline comment a docstring instead of a comment).  Please reword as you see fit, just a suggested wording
```suggestion
async def test_profile_resource_creation(lightkube_client, profile):
    """Test whether resources expected for a profile were created."""
```

Reply:
oh wait I just misunderstood the test.  You are actually testing whether the `Profile`'s stuff was created here, not whether the things you create (secrets, etc) are present

Reply:
This is more subjective, but I think this testing functionality would make more sense inside the 'profile' fixture. My gut says tests here test things your charm is doing, and the fixtures are for providing stuff that comes from elsewhere, so seeing a fixture fail would indicate something broke in my dependencies. Having said that though, there's some flexibility here.

I'm pretty sure pytest-operator runs tests linearly so you can always guarantee this test runs before anything below it, but putting the test/raise if it breaks functionality into the fixture itself might make it more clear